mongodbatlas/resource_mongodbatlas_federated_database_instance.go

```diff
@@ -747,91 +728,89 @@ func flattenDataFederationDatabase(atlasDatabases []*matlas.DataFederationDataba
 	return dbs
 }
 
-func flattenDataFederationDatabaseViews(atlasViews []*matlas.DataFederationDatabaseView) []map[string]interface{} {
+func flattenDataFederationDatabaseViews(atlasViews []admin.DataLakeApiBase) []map[string]interface{} {
 	views := make([]map[string]interface{}, len(atlasViews))
 
 	for i, atlasView := range atlasViews {
 		views[i] = map[string]interface{}{
-			"name":     atlasView.Name,
-			"source":   atlasView.Source,
-			"pipeline": atlasView.Pipeline,
+			"name":     atlasView.GetName(),
+			"source":   atlasView.GetSource(),
+			"pipeline": atlasView.GetPipeline(),
 		}
 	}
 
 	return views
 }
 
-func flattenDataFederationCollections(atlasCollections []*matlas.DataFederationCollection) []map[string]interface{} {
+func flattenDataFederationCollections(atlasCollections []admin.DataLakeDatabaseCollection) []map[string]interface{} {
```

Comment:
```suggestion
func flattenDataFederationCollections(atlasCollections []*admin.DataLakeDatabaseCollection) []map[string]interface{} {
```

Reply:
`atlasCollections []admin.DataLakeDatabaseCollection` is the data type returned by the SDK, unable to use pointers as above